code/game/machinery/doors/door.dm

```diff
@@ -76,8 +76,14 @@
 		var/mob/living/M = AM
 		if(world.time - M.last_bumped <= 10) return	//Can bump-open one airlock per second. This is to prevent shock spam.
 		M.last_bumped = world.time
-		if(!M.restrained() && M.mob_size > MOB_SIZE_SMALL)
-			bumpopen(M)
+		if(!M.restrained())
+			if(ispet(AM))
+				var/mob/living/simple_animal/A = AM
+				if(A.collar)

```

Comment:
I'd prefer to check if both are true in one statement, but I can't see a way of doing that, unfortunately. Thoughts?

Reply:
The colon operator is generally considered "avoid this" by the coding standards we try to put in place though. While it does work and has cases where it is somewhat unavoidable, the maintainers will lament loudly and gnash their teeth at the sign of new instances of its use.

It is up there with relative pathing and `src.` when it comes to how much they hate it.